b2sdk/exception.py

```diff
@@ -556,6 +556,22 @@ class PotentialS3EndpointPassedAsRealm(InvalidJsonResponse):
     pass
 
 
+class DestinationDirectoryError(B2Error):
+    pass
+
+
+class DestinationDirectoryDoesntExist(DestinationDirectoryError):
+    pass
+
+
+class DestinationIsADirectory(DestinationDirectoryError):
+    pass
+
+
+class DestinationDirectoryDoesntAllowOperation(DestinationDirectoryError):
+    pass
```

Comment:
"Windows doesn't raise IsADirectoryError, raises PermissionError instead" is an interesting piece of information that should be stored in the docstring of this exception. Also the other exceptions should note that so that the windows users thinking they will get DestinationIsADirectory will know they will probably get DestinationDirectoryDoesntAllowOperation instead

Reply:
I've modified the behaviour so it's consistent on all platforms. I figured that we shouldn't provide an interface that's on a whim of operating system. Instead of just checking the exception, I'm actually checking path and parent path and raise most of the exceptions basing on that info.

I've also added an exception for a case when the parent is not a directory, but e.g. symbolic link or a file.